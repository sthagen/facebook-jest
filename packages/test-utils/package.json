{
  "name": "@jest/test-utils",
  "version": "25.3.0",
  "private": true,
  "license": "MIT",
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "typesVersions": {
    "<3.8": {
      "build/*": [
        "build/ts3.4/*"
      ]
    }
  },
  "dependencies": {
    "@types/jest": "*",
    "@types/node": "*",
    "@types/semver": "^6.2.1",
    "ansi-regex": "^5.0.0",
    "ansi-styles": "^4.1.0",
    "semver": "^6.3.0"
  }
}
