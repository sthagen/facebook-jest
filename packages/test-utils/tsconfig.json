{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "build",
    // we don't want `@types/jest` to be referenced
    "types": ["node"]
  },
  "include": ["./src/**/*"],
  "references": [{"path": "../jest-types"}, {"path": "../pretty-format"}]
}
