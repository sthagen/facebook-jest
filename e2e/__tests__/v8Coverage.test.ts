/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as path from 'path';
import {onNodeVersions} from '@jest/test-utils';
import runJest from '../runJest';

const DIR = path.resolve(__dirname, '../v8-coverage');

onNodeVersions('>=10', () => {
  test('prints coverage', () => {
    const sourcemapDir = path.join(DIR, 'no-sourcemap');

    const {stdout, exitCode} = runJest(
      sourcemapDir,
      ['--coverage', '--coverage-provider', 'v8'],
      {
        stripAnsi: true,
      },
    );

    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
  });
});
