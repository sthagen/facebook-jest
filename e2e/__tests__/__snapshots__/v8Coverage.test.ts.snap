// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on node >=10 prints coverage 1`] = `
"  console.log
    42

      at Object.log (__tests__/Thing.test.js:10:9)

----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |     100 |      100 |     100 |     100 |                   
 Thing.js |     100 |      100 |     100 |     100 |                   
 x.css    |     100 |      100 |     100 |     100 |                   
----------|---------|----------|---------|---------|-------------------"
`;
